core/src/main/scala/com/salesforce/op/stages/impl/selector/ModelSelector.scala

```diff
@@ -156,7 +158,11 @@ E <: Estimator[_] with OpPipelineStage2[RealNN, OPVector, Prediction]]
       best
     }
 
-    val preparedData = splitter.map(_.validationPrepare(datasetWithID)).getOrElse(datasetWithID)
+    val preparedData = splitter.collect {
```

Comment:
I find this part quite confusing. We apply `validationPrepare` only if it's a `DataCutter`, and then only if it's not `DataCutter` we apply `validationPrepare`. So what I get from this is that when it's a `DataCutter` we want to apply `validationPrepare` early, but not for any other splitter types. Perhaps it's worth adding a few lines of docs explaining why.